public/app/features/variables/types.ts

```diff
@@ -119,11 +119,32 @@ export interface SystemVariable<TProps extends { toString: () => string }> exten
   current: { value: TProps };
 }
 
+export enum VariableLoadingState {
+  NotStarted = 'Not started',
+  Fetching = 'Fetching',
+  Completed = 'Completed',
+  Failed = 'Failed',
```

Comment:
Can we use the LoadingState enum from grafana/data ? , only one that is not applicable is Streaming

Reply:
Great suggestion, will lock into that